ce/ce/cli/commands/find.ts

```diff
@@ -42,14 +43,30 @@ export class FindCommand extends Command {
     const table = new Table('Artifact', 'Version', 'Summary');
 
     for (const each of this.inputs) {
-      for (const [registry, id, artifacts] of await registries.search({ keyword: each, version: this.version.value })) {
-        const latest = artifacts[0];
-        if (!latest.metadata.info.dependencyOnly) {
-          const name = artifactIdentity(latest.registryId, id, latest.shortName);
-          table.push(name, latest.metadata.info.version, latest.metadata.info.summary || '');
+      // eslint-disable-next-line prefer-const
+      for (let [registry, id, artifacts] of await registries.search({
+        // use keyword search if no registry is specified
+        keyword: each.indexOf(':') === -1 ? each : undefined,
+        // otherwise use the criteria as an id
+        idOrShortName: each.indexOf(':') > -1 ? each : undefined,
+        version: this.version.value
+      })) {
+        if (!this.version.isRangeOfVersions) {
+          // if the user didn't specify a range, just show the latest version that was returned
+          artifacts = [artifacts[0]];
+        }
+        for (const result of artifacts) {
+          if (!result.metadata.info.dependencyOnly) {
+            const name = artifactIdentity(result.registryId, id, result.shortName);
+            table.push(name, result.metadata.info.version, result.metadata.info.summary || '');
+          }
         }
       }
     }
+    if (!table.anyRows) {
```

Comment:
Why is this a member of table rather than just a local here?

(Not asking to change it, asking to understand myself)

Reply:
I figured that it was reusable; but not important to be so.